idioms/fallible-closure.md

```diff
@@ -0,0 +1,35 @@
+# Fallible closure inside infallible function
+
+## Description
+
+Build a fallible closure inside of an apparently infallible function.
+
+## Example
+
+```rust
+use std::path::Path;
+
+// this function hides its failure states
+fn file_stem(path: &Path) -> &str {
+    // this closure is fallible, so it can make use of `?`
+    // fn() -> Option<&str>
+    let inner = || Some(path.file_stem()?.to_str()?);
+    inner().unwrap_or("untitled")
+}
+```
+
+## Motivation
+
+You may sometimes want to use the [`?`] operator inside an infallible function,
+in those cases a closure inside your function could be the simple solution.
+
+## Advantages
+
+This allows using the terser syntax permitted by the [`?`] operator inside an
+infallible context.
+
+## Disadvantages
+
+This is hiding failure states from the consumer of the API.
```

Comment:
why is that bad and when do you really want to avoid it? mentioning some cases would be good and probably also state, that if this is overused it could be considered an `anti-pattern`